packages/discord.js/typings/index.d.ts

```diff
@@ -2429,13 +2429,13 @@ export class MessagePayload {
 
 export class MessageReaction {
   private constructor(client: Client<true>, data: RawMessageReactionData, message: Message);
-  private _emoji: GuildEmoji | ReactionEmoji;
+  private _emoji: GuildEmoji | ReactionEmoji | ApplicationEmoji;
 
   public burstColors: string[] | null;
   public readonly client: Client<true>;
   public count: number;
   public countDetails: ReactionCountDetailsData;
-  public get emoji(): GuildEmoji | ReactionEmoji;
+  public get emoji(): GuildEmoji | ReactionEmoji | ApplicationEmoji;
```

Comment:
In its current state, it is not possible for `MessageReaction#emoji` along with its associated private property to return an ApplicationEmoji, which would make these typings inaccurate.

Reply:
I think it makes sense for a MessageReaction done by the bot with an ApplicationEmoji to return ApplicationEmoji for its emoji property. Do you know where I can make these changes?

Reply:
Since application emojis need to be fetched to be had, I wouldn't actually try to squeeze that in. it's fine to leave it as before

Reply:
why not add a cache check for ApplicationEmoji here?
https://github.com/discordjs/discord.js/blob/4594896b5404c6a34e07544951c59ff8f3657184/packages/discord.js/src/structures/MessageReaction.js#L126

Reply:
Oof, _that_ getter. I'd maybe add that logic into the getter to store app emojis if we *really* wanna go there

Reply:
I have added `ApplicationEmoji` lookup to the getter.

Reply:
That now changes the scope of the PR to not be types only. Please edit the PR title and description accordingly and make sure you actually tested that change.